src/backend/metal/src/device.rs

```diff
@@ -276,29 +303,40 @@ impl Device {
         pipeline.set_input_primitive_topology(primitive_class);
 
         // Shaders
-        let vs_lib = match pipeline_desc.shaders.vertex.module {
-            &n::ShaderModule::Compiled(ref lib) => lib.to_owned(),
+        let (vs_lib, vs_remapped_entries) = match pipeline_desc.shaders.vertex.module {
+            &n::ShaderModule::Compiled {ref library, ref remapped_entry_point_names} => (library.to_owned(), remapped_entry_point_names.to_owned()),
```

Comment:
why do we need to clone the entry points here?

Reply:
The lifetime of the hashmap returned in the `Raw` branch is too short, because I want to return a reference in both `match` branches. Take a look at my new commit to see my issue, this avoids the clone but it feels messy. Is there a way to do this without the temporary `_owned` variables?